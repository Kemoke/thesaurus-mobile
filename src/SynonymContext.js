import React from 'react';

export type SynonymContextProps = {
    words: Map<String, Array<string>>,
    addSynonym: (word: string, synonym: string) => void
}

export const SynonymContext = React.createContext({
    words: {},
    addSynonym: (word, synonym) => {}
});

export const SynonymContextWrapper = (Component) => (props) => (
    <SynonymContext.Consumer>
        {ctx => (
            <Component {...props} synonymContext={ctx}/>
        )}
    </SynonymContext.Consumer>
);