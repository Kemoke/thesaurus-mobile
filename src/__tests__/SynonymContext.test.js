import React from 'react';
import {shallow} from "enzyme";
import {SynonymContextWrapper} from "../SynonymContext";
import {View} from "react-native";

describe('SynonymContextWrapper', function () {
    it('should wrap component properly', function () {
        const WrappedView = SynonymContextWrapper(View);
        const component = shallow(
            <WrappedView testProp="test"/>
        );
        expect(component).toMatchSnapshot();
        const context = {
            words: {test: ["test"]}
        };
        const wrappedComponent = shallow(component.props().children(context));
        expect(wrappedComponent).toMatchSnapshot();
    });
});