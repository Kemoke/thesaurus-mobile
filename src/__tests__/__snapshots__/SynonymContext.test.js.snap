// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SynonymContextWrapper should wrap component properly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`SynonymContextWrapper should wrap component properly 2`] = `
<View
  synonymContext={
    Object {
      "words": Object {
        "test": Array [
          "test",
        ],
      },
    }
  }
  testProp="test"
/>
`;
